packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -375,12 +339,66 @@ export default defineComponent({
       return l1Direct ? -1 : 1
     },
 
+    checkPasswordEnforcedFor(link) {
+      const currentRole = this.availableRoleOptions.find(({ role }) => {
+        return parseInt(link.permissions) === role.bitmask(false)
+      })
+
+      const canRead = currentRole.role.hasPermission(SharePermissions.read)
+      const canCreate = currentRole.role.hasPermission(SharePermissions.create)
+      const canDelete = currentRole.role.hasPermission(SharePermissions.delete)
+
+      if (this.passwordEnforced.read_only === true) {
+        return canRead && !canCreate && !canDelete
+      }
+      if (this.passwordEnforced.upload_only === true) {
+        return !canRead && canCreate && !canDelete
+      }
+      if (this.passwordEnforced.read_write === true) {
+        return canRead && canCreate && !canDelete
+      }
+      if (this.passwordEnforced.read_write_delete === true) {
+        return canRead && canCreate && canDelete
+      }
+      return false
+    },
+
     addNewLink() {
-      this.currentView = VIEW_CREATE
+      this.checkLinkToCreate({
+        link: {
+          name: this.$gettext('Link'),
+          role: this.availableRoleOptions[0],
+          permissions: 1,
+          expiration: this.expirationDate.default,
+          password: false
+        }
+      })
     },
 
-    cancelCreation() {
-      this.currentView = VIEW_SHOW
+    checkLinkToCreate({ link, onError = () => {} }) {
+      const paramsToCreate = this.getParamsForLink(link)
+
+      if (this.checkPasswordEnforcedFor(link)) {
+        showQuickLinkPasswordModal({ store: this.$store }, async (newPassword) => {
+          this.hideModal()
+          this.createLink({ params: { ...paramsToCreate, password: newPassword }, onError })
+        })
+      } else {
+        this.createLink({ params: paramsToCreate, onError })
+      }
+    },
+
+    checkLinkToUpdate({ link, onSuccess = () => {} }) {
+      const params = this.getParamsForLink(link)
+
+      if (this.checkPasswordEnforcedFor(link)) {
```

Comment:
huh? this will enforce to set a new password on any edit action of a link, no?

Reply:
Shouldn't, let me double-check. But the idea was to check if passwords are enforced and if yes show the modal (and then progress in the update progress upon password submit) and if no then directly progress with the update

Upon second look we need to check _if the already has a password_, too. Not sure if you were hinting at that?

Reply:
True yeah, I've added a mitigation. Thanks!